src/planner/binder/expression/bind_columnref_expression.cpp

```diff
@@ -255,23 +254,24 @@ BindResult ExpressionBinder::BindExpression(ColumnRefExpression &colref_p, idx_t
 		return BindResult(make_unique<BoundConstantExpression>(Value(LogicalType::SQLNULL)));
 	}
 	string error_message;
-	auto expr = QualifyColumnName(colref_p, error_message);
-	if (!expr) {
+	auto qualify_column_result = QualifyColumnName(colref_p, error_message);
+	if (!qualify_column_result.expression) {
 		return BindResult(binder.FormatError(colref_p, error_message));
 	}
-	expr->query_location = colref_p.query_location;
+	qualify_column_result.expression->query_location = colref_p.query_location;
 
 	// a generated column returns a generated expression, a struct on a column returns a struct extract
-	if (expr->type != ExpressionType::COLUMN_REF) {
-		auto alias = expr->alias;
-		auto result = BindExpression(&expr, depth);
+	if (qualify_column_result.expr_is_generated ||
```

Comment:
That's fair, I figured since we already needed the error string everywhere and we were returning a ParsedExpression, bundling them would be nice and allowed me to communicate that the columnref was a generated column.

The reason for this change is that the expression can also just be a regular columnref, e.g `c1 as c0`
That would not get picked up by the previous condition, though I did not encounter an issue caused by that behavior, so maybe it's not needed at all

Reply:
I think the `ParseResult` looks good, although I would just (1) add implicit casting from `unique_ptr<ParsedExpression>` (so we don't need to specify `ParseResult` everywhere) and (2) remove `string &error_message` from all functions (as the parse result should mean we don't need that anymore).

I would also remove the `expr_is_generated` from the `ParseResult` and rather have a check here that checks if the column reference is generated - that seems less error prone to me as then we cannot accidentally forget to set the flag in a certain branch.